segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogFactory.java

```diff
@@ -142,7 +142,9 @@ private BookKeeper startBookKeeperClient() throws Exception {
                 .setGetBookieInfoTimeout(readTimeout)
                 .setEnableDigestTypeAutodetection(true)
                 .setClientConnectTimeoutMillis((int) this.config.getZkConnectionTimeout().toMillis())
-                .setZkTimeout((int) this.config.getZkConnectionTimeout().toMillis());
+                .setZkTimeout((int) this.config.getZkConnectionTimeout().toMillis())
+                .setNettyMaxFrameSizeBytes(BookKeeperConfig.MAX_APPEND_LENGTH)
```

Comment:
This value is smaller than the actual frame length. The BK Client will wrap our data in an envelope which will increase this size. If this value is passed per-se to Netty, I don't understand how the system could still work. That's unless there's other code in the BK client to increase this value to account for the extra envelope size. 

Would you mind checking in the BK client code to see how this is handled?